lib/ohai/config.rb

```diff
@@ -16,28 +18,93 @@
+require 'ohai/log'
+require 'chef-config/logger'
+
+ChefConfig::Logger = Ohai::Log
+
+require 'chef-config/config'
 
 module Ohai
+  Config = ChefConfig::Config
+
+  # Reopens ChefConfig::Config to add Ohai configuration settings.
+  # see: https://github.com/chef/chef/blob/master/lib/chef/config.rb
   class Config
-    extend Mixlib::Config
-
-    # from chef/config.rb, should maybe be moved to mixlib-config?
-    def self.platform_specific_path(path)
-      if RUBY_PLATFORM =~ /mswin|mingw|windows/
-        # turns /etc/chef/client.rb into C:/chef/client.rb
-        path = File.join(ENV['SYSTEMDRIVE'], path.split('/')[2..-1])
-        # ensure all forward slashes are backslashes
-        path.gsub!(File::SEPARATOR, (File::ALT_SEPARATOR || '\\'))
-      end
-      path
+    # These methods need to be defined before they are used as config defaults,
+    # otherwise they will get method_missing'd to nil.
+    private
+    def self.default_hints_path
+      [ ChefConfig::Config.platform_specific_path('/etc/chef/ohai/hints') ]
+    end
+
+    def self.default_plugin_path
+      [ File.expand_path(File.join(File.dirname(__FILE__), 'plugins')) ]
     end
 
+    public
+    # Copy deprecated configuration options into the ohai config context.
+    def self.merge_deprecated_config
+      [ :hints_path, :plugin_path ].each do |option|
+        if has_key?(option) && send(option) != send("default_#{option}".to_sym)
+          warn_deprecated(option)
+        end
+      end
 
+      ohai.merge!(configuration)
+    end
+
+    # Keep "old" config defaults around so anyone calling Ohai::Config[:key]
+    # won't be broken. Also allows users to append to configuration options
+    # (e.g., Ohai::Config[:plugin_path] << some_path) in their config files.
+    default :disabled_plugins, []
+    default :hints_path, default_hints_path
     default :log_level, :info
     default :log_location, STDERR
-    default :plugin_path, [ File.expand_path(File.join(File.dirname(__FILE__), 'plugins'))]
-    default :disabled_plugins, []
-    default(:hints_path) { [ platform_specific_path('/etc/chef/ohai/hints') ] }
+    default :plugin_path, default_plugin_path
+
+    # Log deprecation warning when a top-level configuration option is set.
+    # TODO: Should we implement a config_attr_reader so that deprecation
+    # warnings will be generatd on read?
+    [
+      :directory,
+      :disabled_plugins,
+      :log_level,
+      :log_location,
+      :logfile, # TODO: Listed in link above but only seen in application.rb
+      :version
+    ].each do |option|
+      # https://docs.chef.io/config_rb_client.html#ohai-settings
+      # hints_path and plugin_path are intentionally excluded here; warnings for
+      # setting these attributes are generated in merge_deprecated_config since
+      # append (<<) operations bypass the config writer.
+      config_attr_writer option do |value|
+        warn_deprecated(option)
+        value
+      end
+    end
```

Comment:
We will need to ensure that ohai config can process `:log_level` and `:log_location` as it does now (e.g., sets `Ohai.config[:log_level]` if `:log_level` is configured)